src/components/application_manager/src/commands/mobile/perform_audio_pass_thru_request.cc

```diff
@@ -343,6 +343,30 @@ void PerformAudioPassThruRequest::FinishTTSSpeak(){
   SendHMIRequest(hmi_apis::FunctionID::TTS_StopSpeaking, NULL);
 }
 
+bool PerformAudioPassThruRequest::WaitTTSSpeak() {
+  LOG4CXX_AUTO_TRACE(logger_);
+  uint32_t default_timeout_msec =
+      profile::Profile::instance()->default_timeout();
+  TimevalStruct start_time = date_time::DateTime::getCurrentTime();
+
+  // Waiting for TTS_Speak
+  while (is_active_tts_speak_) {
+    int64_t difference_between_start_current_time
+        = date_time::DateTime::calculateTimeSpan(start_time);
+    // Send GENERIC_ERROR after default timeout
+    if (difference_between_start_current_time > default_timeout_msec) {
+      LOG4CXX_ERROR(logger_, "Expired timeout for TTS.Speak response");
```

Comment:
ERROR is something big as a problem.
It looks more like WARNING - non-casual, but expected behaviour. Or INFO if it real casual situation (for example it depends on user actions)